### PR TITLE
Add 'nlohmann_fifo_map' package

### PR DESCRIPTION
**Pull request by rbsheth**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/create/cmake.html)
  step by step carefully. **Yes**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/rbsheth/hunter/builds/21000809
  * https://travis-ci.com/rbsheth/hunter/builds/94724804

* I have submitted CI configs to https://github.com/ingenue/hunter targeting `pkg.template` branch,
  see this merged pull request https://github.com/ingenue/hunter/pull/309

* My change will resolve this "New package" request: https://github.com/ruslo/hunter/issues/1652

### CODE REVIEW

File: docs/packages/pkg/nlohmann_fifo_map.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.. spelling::
+
+    nlohmann
+    fifo_map
```
Comment (ruslo) [inline docs/packages/pkg/nlohmann_fifo_map.rst]: Error: https://travis-ci.org/ruslo/hunter/builds/469294600#L2474

Should be:
```
 .. spelling::
 
    nlohmann
    fifo
    map
```

Comment (rbsheth) [reply docs/packages/pkg/nlohmann_fifo_map.rst]: done

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/builds/21078437
* https://travis-ci.org/ingenue/hunter/builds/469305647?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/builds/21078756
* https://travis-ci.org/ingenue/hunter/builds/469311239?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.76

**Pull request merged**